quarkus/maven-plugin/pom.xml

```diff
@@ -0,0 +1,62 @@
+
+<!--
+  ~ Copyright 2016 Red Hat, Inc. and/or its affiliates
+  ~ and other contributors as indicated by the @author tags.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <parent>
+        <artifactId>keycloak-docs-parent</artifactId>
+        <groupId>org.keycloak</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../../operator/pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <name>Keycloak Operator Maven Plugin</name>
+    <artifactId>keycloak-operator-maven-plugin</artifactId>
+    <description>Keycloak Operator Maven Plugin</description>
+    <packaging>maven-plugin</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-plugin-api</artifactId>
+            <version>3.6.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven.plugin-tools</groupId>
+            <artifactId>maven-plugin-annotations</artifactId>
+            <version>3.6.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-project</artifactId>
+            <version>2.2.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-quarkus-server</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.github.javaparser</groupId>
+            <artifactId>javaparser-core</artifactId>
```

Comment:
The storage is also generating sources and using a different approach.

In the past, we had some discussions about using a 3rd party for generating sources. See https://groups.google.com/g/keycloak-dev/c/MhLD-YoKVCs/m/e5FcDtejAgAJ. See some arguments from Hynek about why he is not using any 3rd party library.

Reply:
For the seek of discussion I report here the relevant comment:

> A few frameworks have been considered for the class generation, be it on source or byte-code level -  lombok, autovalue, or bytebuddy to name a few. They have proven to be unnecessarily heavy and also by adding extra dependencies, we'd add an extra maintenance cost.

I do believe that this doesn't apply to this PR:
 - here the use-case is to generate plain `java` files and not byte-code
 - `javaparser` is not heavy at all
 - `javaparser` offers a simple and intuitive interface
 - `javaparser` generated code is syntactically checked with nice error handling and relevant Exceptions
 - is not a new and extra dependency, since it's already on the classpath in the operator at least (`crd-generator` -> `sundrio` -> `javaparser`)
 - is proven to be an effective and widely accepted tool to generate java code

Reply:
It does apply because one of the arguments from Hynek is not using any 3rd party library to avoid adding extra dependencies too.

As you can see from that thread, I am also in favor of using a library like `jdeparser`.

IMO, if we can align the solution and use the same approach for generating sources across the code base, better. If others changed their mind and using `jdeparser` is better, even better.

Reply:
Let's discuss this subject when we have more visibility about the desired outcome.

Regarding the discussion about code generation itself I summarize the outcome of my discussion with @pedroigor (please, feel free, to correct me if I took anything wrongly):

 - we agree that generating plain Java code (not byte-code as stated in the google groups discussion) using a library is a more solid foundation and less painful for development/maintenance as opposed to plain string interpolation (which is the current approach from Store.X team AFAIK)
 - we observed that, as of today, both `javaparser` and `jdeparser` are both, in some way, transitively used in this repository. Since they are anyhow dependencies, the argument of adding extra maintenance due to the extra dependency doesn't hold
 - `jdeparser` is much less used than `javaparser` in the wild even if it's a RH project
 - both `jdeparser` and `javaparser` are used in the Quarkus project for different purposes
 - I do have a very good experience using `javaparser` [in other projects](https://github.com/fabric8io/kubernetes-client/tree/master/java-generator)
 
 > They are facing very similar issues AFAIK.
 
 Not sure about this point, let's see where we end up going with this, but generating internals might be a very much different concern than generating public APIs.

Reply:
@andreaTP Thanks for providing more context.

Reply:
Any new dependency has to be productized before being merged. The higher the number of (direct or indirect) dependencies, the higher cost due to maintenance of CVE fixes.

Can you provide the list of all the dependencies brought in via this dependency?